app/controllers/artefacts_controller.rb

```diff
@@ -210,7 +210,7 @@ def extract_parameters(params)
         "sections" => [],
         "specialist_sectors" => [],
         "related_artefact_slugs" => [],
-        "external_links_attributes" => [:title, :url, :_id, :_destroy],
+        "external_links_attributes" => [:title, :url, :id, :_destroy],
```

Comment:
This array is used to permit the params from the form. The rails `fields_for` helper adds a hidden input with the name `id` when the the object, in this case an `ArtefactExternalLink` is rendered to the page. This field (along with `_destroy = true | false` gets submitted back with the form and is used by `accepts_nested_attributes_for` to determine what action to take. So... Previously, the `id` parameter was being filtered out resulting in every form submission creating a new `ArtefactExternalLink` (or loads of them). If you add an external link on an artefact on integration and then keep saving the page you'll see them 'multiply'.